### PR TITLE
Update hover functionality in Chip component

### PR DESCRIPTION
**Pull request by ksmort**
## Description
Add section to Chip page for `onClick` prop.
Update the icon section to use `icon` prop instead of `iconStart`.
Added in the hover support.

## Notes
Not 100% on what colour we want the hover to be, maybe checkout the branch and do a look?
Also a TODO: convert to function component.

## Screenshots
Screenshot_2020-12-07 Chip Riipen UI

## Where to Start
91 packages/riipen-ui/src/components/Chip.jsx

### CODE REVIEW

File: packages/riipen-ui-docs/src/pages/components/chip/Hover.jsx [hunk 1]
```diff
@@ -28,6 +28,26 @@ export default function Hover() {
         <Chip label="Warning" hover color="warning" />
         <span style={style} />
         <Chip label="Negative" hover color="negative" />
+        <span style={style} />
```
Comment (ksmort) [inline packages/riipen-ui-docs/src/pages/components/chip/Hover.jsx:31]: This file already existed, so I'm going to assume hover support was supposed to exist, but maybe got forgotten. :)

File: packages/riipen-ui/src/components/Chip.jsx [hunk 2]
```diff
@@ -198,85 +209,137 @@ class Chip extends React.Component {
             border-color: ${theme.palette.primary.main};
             color: ${theme.palette.primary.contrast};
           }
+          .primary.hover:hover {
```
Comment (ksmort) [inline packages/riipen-ui/src/components/Chip.jsx:212]: Not sure if we want to darken chips on hover, thoughts?

File: packages/riipen-ui/src/components/Chip.jsx [hunk 3]
```diff
@@ -198,85 +209,137 @@ class Chip extends React.Component {
...
+            background-color: ${theme.palette.positive.dark};
+            border-color: ${theme.palette.positive.dark};
+          }
           .positive.outlined {
             border-color: ${theme.palette.positive.main};
             color: ${theme.palette.positive.main};
           }
-
+          .positive.outlined.hover:hover {
+            background-color: transparent;
```
Comment (ksmort) [inline packages/riipen-ui/src/components/Chip.jsx:312]: Also not sure on colour change for outlined version.

File: packages/riipen-ui/src/components/Chip.jsx (hunk 2 above)
Comment (bolinkd) [reply packages/riipen-ui/src/components/Chip.jsx:212]: I was thinking that _outlined_ chips would become _default_ and _default_ would become _outlined_. 

But that also depends on what the designs specify, I would ask Maribel to decide what she wants to do in this situation

Comment (ksmort) [reply packages/riipen-ui/src/components/Chip.jsx:212]: I shall ask her, convert this PR to draft for now, and then update when ready!

Comment (ksmort) [reply packages/riipen-ui/src/components/Chip.jsx:212]: Updated to be the inverted variant!

**Review by bolinkd — APPROVED** [review]

**Pull request merged**